# Large-scale fungal strain sequencing unravels the molecular diversity in mating loci maintained by long-term balancing selection (TriMAT)
**Authors**: David Peris, Dabao Sun Lu, Vilde Bruhn Kinneberg, Ine-Susanne Methlie, Malin Stapnes Dahl, Timothy Y. James, Håvard Kauserud, Inger Skrede

**Journal**: Plos Genetics 18(3): e1010097

**Year**: 2022

**Abstract**: Balancing selection, an evolutionary force that retains genetic diversity, has been detected in multiple genes and organisms, such as the sexual mating loci in fungi. However, to quantify the strength of balancing selection and define the mating-related genes require a large number of strains. In tetrapolar basidiomycete fungi, sexual type is determined by two unlinked loci, MATA and MATB. Genes in both loci define mating type identity, control successful mating and completion of the life cycle. These loci are usually highly diverse. Previous studies have speculated, based on culture crosses, that species of the non-model genus Trichaptum (Hymenochaetales, Basidiomycota) possess a tetrapolar mating system, with multiple alleles. Here, we sequenced a hundred and eighty strains of three Trichaptum species. We characterized the chromosomal location of MATA and MATB, the molecular structure of MAT regions and their allelic richness. The sequencing effort was sufficient to molecularly characterize multiple MAT alleles segregating before the speciation event of Trichaptum species. Analyses suggested that long-term balancing selection has generated trans-species polymorphisms. Mating sequences were classified in different allelic classes based on an amino acid identity (AAI) threshold supported by phylogenetics. 17,550 mating types were predicted based on the allelic classes. In vitro crosses allowed us to support the degree of allelic divergence needed for successful mating. Even with the high amount of divergence, key amino acids in functional domains are conserved. We conclude that the genetic diversity of mating loci in Trichaptum is due to long-term balancing selection, with limited recombination and duplication activity. The large number of sequenced strains highlighted the importance of sequencing multiple individuals from different species to detect the mating-related genes, the mechanisms generating diversity and the evolutionary forces maintaining them. 

This repository, linked to the TriMAT [Github Page](https://perisd.github.io/TriMAT/ "TriMAT Webpage"), contains some of the raw data used to generate Figures and Tables in this publication. Additional raw data can be found in dryad repository DOI: [Dryad repository](https://doi.org/10.5061/dryad.fxpnvx0t4 "Dryad repository")

**[MAT dryad repository](https://doi.org/10.5061/dryad.fxpnvx0t4 "MAT dryad")**

Note:

Supplementary Figure 13 (Figure S13) was generated using the proteins retrieved from JGI genomes:
1. 1KFG: Fomitiporia mediterranea, Onnia scaura, Phellinidium ferrugineofuscum, Phellinus igniarius, Phellinus viticola, Phellopilus nigrolimitatus, Resinicium bicolor, Sidera vulgaris.
2. 1000 Fungal Genome Project: Porodaedalea chrysoloma, Porodaedalea niemelaei.
3. Published genomes:
    
    3.1 Rickenella fibula: [Korotkin HB et al. 2018](https://doi.org/10.1002/ajb2.1183 "Korotkin, H. B., et al., "Stable isotope analyses reveal previously unknown trophic mode diversity in the Hymenochaetales," Am J Bot 105, 1869-1887 (2018)")
    
    3.2 Rickenella mellea: [Krizsán, K., et al. 2019](https://www.sciencedirect.com/science/article/pii/S0168165615300407 " Krizsán, K., et al., "Transcriptomic atlas of mushroom development reveals conserved genes behind complex multicellularity in fungi," Proc. Natl. Acad. Sci. U. S. A. 116, 7409-7418 (2019)")
    
    3.3 Schizopora paradoxa: [Min, B., et al. 2015](https://doi.org/10.1073/pnas.1817822116 "Min, B., et al., "Genome sequence of a white rot fungus Schizopora paradoxa KUC8140 for wood decay and mycoremediation," J. Biotechnol 211, 42-43 (2015)")

***

### Contact

[<img src="https://www.uv.es/perisnav/Index/twitter-logo.png" width="50"/>](https://linktr.ee/PerisD)


